lib/puppet/provider/package/yum.rb

```diff
@@ -156,6 +156,11 @@ def install
     operation = :install
 
     case should
+    when :latest
```

Comment:
@petems (Re contributor summit) I figured out where this is going wrong! :D 

We had `when :latest && current_package`, but since this is a case statement, that will always evaluate to false. If we turn this into an if statement instead, this will work with your new tests. See also comment on test about what needs to change there.